sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -558,9 +486,13 @@ void peekMessages(MessagingEntityType entityType, boolean isSessionEnabled) {
                 .assertNext(message -> checkCorrectMessage.accept(message, 7))
                 .verifyComplete();
         } finally {
-            receiveAndDeleteReceiver.receiveMessages()
-                .take(messages.size())
-                .blockLast(Duration.ofSeconds(15));
+            StepVerifier.create(receiver.receiveMessages().take(messages.size()))
+                .thenConsumeWhile(receivedMessage -> {
```

Comment:
Then consume while always returns true? Shouldn't we assert that `completed < messages.size()` and then stop when `completed >= messages.size()`? And instead of thenCancel(), expect a complete?